controllers/admissionpolicy_controller.go

```diff
@@ -66,7 +66,7 @@ func (r *AdmissionPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 // SetupWithManager sets up the controller with the Manager.
 func (r *AdmissionPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	err := ctrl.NewControllerManagedBy(mgr).
-		For(&v1alpha2.AdmissionPolicy{}).
+		For(&policiesv1.AdmissionPolicy{}).
```

Comment:
I haven't tested it yet, I will later today. But I suspect if we only do this, existing `v1alpha2` resources are not going to be reconciled by this controller as it is. Note: if this is an issue, this applies to `ClusterAdmissionPolicy` and `PolicyServer` resources as well in their controllers.

What passive operation is going to "upgrade" the version of the existing objects? The idea in this kind of migration is that it happens passively: an object is read in `v1alpha2`, and the component writes it back after the reconciliation in the default stored version: `v1`, effectively migrating the resource. More info: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version.

So, for this PR, I think we have to also watch `v1alpha2` resources. By updating them in regular reconcile loops, they will get migrated to `v1`.

-- Aside, unrelated to this PR strictly:

For `1.1.0` we will have to:
- Remove the watch for `v1alpha2` resources on the controller.
- Remove the existing `v1alpha2` resources that have been ignored since `1.0`.
- Remove the `v1alpha2` CRD's.

At that point in time, only `v1` resources will exist and will be allowed.

-- /Aside

Reply:
I also thought about that, however it seems to work fine in the local test I did. I created a PolicyServer and a ClusterAdmission  with v1alpha2 using an old version of the controller. Then I updated to the latest version of the controller and the controller could see changes in the the v1alpha2 resources (eg when I deleted the PolicyServer resource everything was removed)

I believe this is because we watch the v1 and k8s returned the v1alpha2 wth the v1 apiVersion:

```
When you read an object, you specify the version as part of the path. If you specify a version that is different from the object's persisted version, Kubernetes returns the object to you at the version you requested, but the persisted object is neither changed on disk, nor converted in any way (other than changing the apiVersion string) while serving the request. You can request an object at any version that is currently served.
```

more info https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#writing-reading-and-updating-versioned-customresourcedefinition-objects

but please do a local test and let me know the outcome, I might be wrong and this is important to get it right. Thanks!

Reply:
I've tested the [upgrade process](https://github.com/kubewarden/kubewarden-end-to-end-tests/pull/27) from an old Helm chart (using `v1alpha2` crd version) to latest one (using `v1` crd version). In my tests, all the `v1alpha2` running objects are migrated to the `v1` and others `v1alpha2` objects installed after the upgrade are stored as `v1` objects.